### PR TITLE
docs: remove guidance to use names

### LINKED ISSUE #21759: Clearly document whether filter names matter
It seems like Envoy does not care about the filter name, but uses the TypeUrl in the config to determine which filter to use.

Relevant commit:
```
commit 8cb6862fe6099cd8583a64ff037ecdeaf0e939fa
Author: Kuat <kyessenov@users.noreply.github.com>
Date:   Thu Apr 7 06:31:49 2022 -0700

    extensions: strongly prefer type URL lookup (#20397)
```

Also discussions in the following issues seem to indicate the same:
https://github.com/envoyproxy/go-control-plane/issues/552
https://github.com/envoyproxy/go-control-plane/issues/293

However, there still exists confusion based on the wording in the docs.

For e.g., the [Envoy doc](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/original_dst_filter) suggests to use the filter name for the OriginalDestination ListenerFilter:
`This filter should be configured with the name envoy.filters.listener.original_dst.`

This contradicts statements about the filter name not being relevant.

Kindly clarify in the docs whether filter names are relevant or not.

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>
Commit Message: make a pass to remove stale suggestion to use filter names.
Risk Level: low
Testing: none
Docs Changes: yes
Release Notes: none
Issue: Fix #21759

### CODE REVIEW

**Review by shashankram — COMMENTED** [review]
Hey @kyessenov, thanks for improving the docs.

Could you clarify if we need to use wellknown names or just have the name match the filter config on the listener when using `typed_per_filter_config` which is a map of filter_name->config?
See https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-route, where for `typed_per_filter_config` it says "The key should match the filter name, such as envoy.filters.http.buffer for the HTTP buffer filter".

**Comment by shashankram** [comment]
> Hey @kyessenov, thanks for improving the docs.
> 
> Could you clarify if we need to use wellknown names or just have the name match the filter config on the listener when using `typed_per_filter_config` which is a map of filter_name->config? See https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-route, where for `typed_per_filter_config` it says "The key should match the filter name, such as envoy.filters.http.buffer for the HTTP buffer filter".

This is actually resulting in an issue described in https://github.com/envoyproxy/envoy/issues/21759#issuecomment-1159243250

**Comment by phlax** [comment]
not sure what the `docs` errors are - something about empty empty

**Comment by phlax** [comment]
@kyessenov wondering what the status is on this one - would be good to update the docs before we cut the next release i think

/wait-any

**Comment by kyessenov** [comment]
@phlax The empty error in docs was because I removed duplicated pages for filters, see how we have two links for some filters:
image

**Comment by kyessenov** [comment]
@phlax Gentle ping.

**Comment by phlax** [comment]
hi - apologies - ive been a bit snowed under - ill review tomorrow

File: docs/root/configuration/http/http_filters/jwt_authn_filter.rst [hunk 1]
```diff
@@ -20,8 +20,6 @@ Following are supported JWT alg:
 Configuration
 -------------
 
-This filter should be configured with the name *envoy.filters.http.jwt_authn*.
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/jwt_authn_filter.rst:23]: this one is missing any api link - which leaves the page a bit of a dead end in terms of configuration

**Review by phlax — COMMENTED** [review]
@kyessenov can we add the new type_url where we are removing the old info ?

many of these pages have incorrect config examples (~unrelatedly) so without the type url they are a bit unhelpful

File: docs/root/configuration/http/http_filters/cache_filter.rst [hunk 2]
```diff
@@ -5,7 +5,6 @@ Cache filter
 
 * :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.cache.v3.CacheConfig>`
 * :ref:`v3 SimpleHTTPCache API reference <envoy_v3_api_msg_extensions.cache.simple_http_cache.v3.SimpleHttpCacheConfig>`
-* This filter should be configured with the name ``envoy.filters.http.cache``.
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/cache_filter.rst:8]: maybe im not understanding something here

```console
$ envoydev/envoy/examples/cache$ git diff
```

```diff
diff --git a/examples/cache/front-envoy.yaml b/examples/cache/front-envoy.yaml
index b953d1e781..caa0ee8b9f 100644
--- a/examples/cache/front-envoy.yaml
+++ b/examples/cache/front-envoy.yaml
@@ -27,8 +27,7 @@ static_resources:
                 route:
                   cluster: service2
           http_filters:
-          - name: "envoy.filters.http.cache"
-            typed_config:
+          - typed_config:
               "@type": "type.googleapis.com/envoy.extensions.filters.http.cache.v3.CacheConfig"
               typed_config:
                 "@type": "type.googleapis.com/envoy.extensions.cache.simple_http_cache.v3.SimpleHttpCacheConfig"
```

```console
$ ./verify.sh
...
front-envoy_1    | [2022-07-13 12:50:15.726][9][critical][main] [source/server/server.cc:117] error initializing configuration '/etc/front-envoy.yaml': Proto constraint validation failed (HttpConnectionManagerValidationError.HttpFilters[0]: embedded message failed validation | caused by HttpFilterValidationError.Name: value length must be at least 1 characters): stat_prefix: "ingress_http"
...
```

ok i see

```diff
diff --git a/examples/cache/front-envoy.yaml b/examples/cache/front-envoy.yaml
index b953d1e781..36bcbcb27b 100644
--- a/examples/cache/front-envoy.yaml
+++ b/examples/cache/front-envoy.yaml
@@ -27,7 +27,7 @@ static_resources:
                 route:
                   cluster: service2
           http_filters:
-          - name: "envoy.filters.http.cache"
+          - name: "my.own.special.filter.name"
             typed_config:
               "@type": "type.googleapis.com/envoy.extensions.filters.http.cache.v3.CacheConfig"

```

...works, hmmm

**Review by phlax — COMMENTED** [review]
@kyessenov 

how about instead of just removing the name advice we replace with something like:

---

At a minimum, this filter must be configured with the following `@type_url` and a name of your choosing:

```yaml
          http_filters:
          - name: "envoy.filters.http.cache"
            typed_config:
              "@type": "type.googleapis.com/envoy.extensions.filters.http.cache.v3.CacheConfig"


```

File: docs/root/configuration/http/http_filters/jwt_authn_filter.rst (hunk 1 above)
Comment (kyessenov) [reply docs/root/configuration/http/http_filters/jwt_authn_filter.rst:23]: Yes, let's put back the type URLs where it's missing. The link is the type URL that should be used generally.

**Review by phlax — APPROVED** [review]
thnks for resolving this @kyessenov - i think this is a really helpful update to the docs, lgtm

**Pull request merged**